### ISSUE TITLE
Ability to enable wizard and other form options in Form Builder

### ISSUE THREAD

**Issue by ebruchez** [orbeon/orbeon-forms#2239]
Currently you have to use properties, which requires a developer or administrator. We would like to make this configurable via a UI in Form Builder.

Properties support global, per app, or per form setting. Here we are only looking at the per form setting.

The UI could be new tab in the Form Settings dialog (BTW form permissions should be moved there too).

The setting could be stored directly in the form definition, very much like form permissions. It could be for example under the `<metadata>` instance.

Other Form Runner properties could be configurable this way, but we can start with `oxf.fr.detail.view.appearance` (which is how the wizard is set) and maybe `oxf.xforms.xbl.fr.wizard.validate`.

See also #2238.

**Comment by ebruchez** [comment]
Since we plan to have a couple of Form Settings properties, maybe we could add this one.

**Comment by ebruchez** [comment]
Other options:

- `xxf:expose-xpath-types`: older forms might have this set to `false`
- `xxf:xpath-analysis`: same
- maybe once we add it: handling of initial calculations on new repeat iterations (unless we do this per grid)

**Comment by ebruchez** [comment]
Some properties are XForms properties, which are placed on `<xf:model>`. This would remain the same.

Where to place properties which are *not* XForms properties, such as `oxf.fr.detail.view.appearance` and `oxf.xforms.xbl.fr.wizard.validate`? Options:

1. on `<xf:model>` in the `fr:` namespace
2. within `<metadata>`

Form Runner currently obtains `oxf.fr.detail.view.appearance` in `components.xsl`, with the `p:property()` function. It could obtain an inline value of the property whether it's on `<xf:model>` or `<metadata>` just as easily.

Are there any other properties for which this wouldn't be the case? For example, Form Runner properties which are needed when the form runs, in which case `<metadata>` would be the better location?

**Comment by ebruchez** [comment]
[+1 from user](https://mail.google.com/mail/u/0/#search/2239/15388502fe5736b0)

**Comment by ebruchez** [comment]
I don't find cases of Form Runner properties which are needed when the form runs, which we would override in the form definition, and which would benefit from being in `<metadata>`.

`<metadata>` is not a bad location either, and allows structured XML, which attributes on `<xf:model>` don't allow. So we could go either way. The Form Runner transform could even update `<metadata>` based on attributes it it came to that, or we could update the property system to adapt.

In short: we just pick a way. Inclined to go for `<metadata>`.

Tasks:

- [ ] Form Builder UI for the wizard setting
- [ ] Form Runner picks property in form when available

**Comment by ebruchez** [comment]
Tentatively added for 2016.2 if customer sponsors.

**Comment by ebruchez** [comment]
One drawback of using `<metadata>` is that we have to filter the value out when writing it to the `metadata` column in `RequestReader`. We should probably change that logic anyway to specify what we want to *include* rather than what we want to *exclude*.

**Comment by ebruchez** [comment]
[Documented](http://doc.orbeon.com/form-runner/feature/wizard-view.html#from-form-builder)
